core/src/main/java/net/opentsdb/query/processor/rate/RateNumericArrayIterator.java

```diff
@@ -95,6 +99,21 @@ public RateNumericArrayIterator(final QueryNode node,
     }
     this.result = result;
     config = (RateConfig) node.config();
+    if (config.getRateToCount()) {
+      if (config.dataIntervalMs() > 0) {
+        data_interval = (config.dataIntervalMs() / 1000) / 
+            config.duration().get(ChronoUnit.SECONDS);
+      } else {
+        // we shouldn't be here without a configured data interval as we don't 
+        // know the underlying data reporting interval and we can't infer it. So
+        // no counts for you!
+        data_interval = 1;
+      }
+      if (data_interval <= 0) {
+        data_interval = 1;
+      }
+    }
```

Comment:
Otherwise we should still set data_interval  to 1?

Reply:
Right we'll assume the count was actually computed earlier. We shouldn't be here unless we're working on a rolled up result and the user passes in valid values. We could validate this in the config in the future.